plugins/window-state/src/lib.rs

```diff
@@ -106,10 +108,12 @@ impl<R: Runtime> WindowExt for Window<R> {
             // otherwise, let the OS decide where to place the window
             for m in self.available_monitors()? {
                 if m.name().map(ToString::to_string).unwrap_or_default() == state.monitor {
-                    self.set_position(PhysicalPosition {
-                        x: state.x,
-                        y: state.y,
-                    })?;
+                    if *m.position() == state.monitor_position && *m.size() == state.monitor_size {
```

Comment:
extract this piece of code into a little helper function `is_last_saved_monitor`